src/components/Profile/ReturningModal/ReturningModal.js

```diff
@@ -64,32 +114,61 @@ class ReturningModal extends Component {
         /** @type {!Array<{componentID:String, quantity: number}>, ...}>}*/
         const currentComponents = this.state.components;
 
-        if (currentComponents[index].quantity > 0)
+        if(currentComponents[index].quantity > 0)
             currentComponents[index].quantity--;
             
         this.setState({ components: currentComponents });
     }
 
     /* 
-    Set the state of the modal and stores user active components 
+    In case of closing the modal, set state to hidden and store components as they were initially 
     */
-    handleShow() { 
-        /** @type {!Array<{componentID:String, quantity: number}>, ...}>}*/
-        const components = ActiveComponents.reservations[this.user_index].activeComponents;
+    handleClose() {
+        if(this.active_user_index !== -1){
+            /** @type {!Array<{componentID:String, quantity: number}>, ...}>}*/
+            const localStorageComponents = JSON.parse(localStorage.getItem('components'));
+            
+            ActiveComponents.reservations[this.active_user_index].activeComponents = localStorageComponents;
+            this.setState({ components: localStorageComponents})
+        }
+        this.setState({ show: false, disabledButton: true });
+    }
 
-        localStorage.setItem('components', JSON.stringify(components));
-        this.setState({ show: true })
+    handleModalList() {
+        /** @type {!Array<{componentID:String, quantity: number}>, ...}>}*/
+        const nextActiveComponents = [];
+        /** @type {!Array<{componentID:String, quantity: number}>, ...}>}*/
+        const localStorageComponents = JSON.parse(localStorage.getItem('components'));
+        
+        this.state.components.forEach((component, index) => {
+            if(this.state.isActive[index] === false) {
+                nextActiveComponents.push(localStorageComponents[index]);
+            } else if(component.quantity < localStorageComponents[index].quantity) {
+                const auxComponent = localStorageComponents[index];
+                auxComponent.quantity -= component.quantity;
+                nextActiveComponents.push(auxComponent);
+            }
+        })
+        this.setState({
+            show: false, 
+            components: nextActiveComponents, 
+            isActive: new Array(nextActiveComponents.length).fill(false), 
+            disabledButton: true 
+        });
+        ActiveComponents.reservations[this.active_user_index].activeComponents = nextActiveComponents;
```

Comment:
This line is also repeated above. Try declaring it as a global variable to avoid repetitions
`this.activeComponents = ActiveComponents.reservations[this.active_user_index].activeComponents`